bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,293 @@
+   */
+  public static BigtableTableAdminSettings createTableAdminSettings(@Nonnull final BigtableOptions options)
+      throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+
+    final BigtableTableAdminSettings.Builder adminBuilder = BigtableTableAdminSettings.newBuilder();
+
+    adminBuilder.setProjectId(options.getProjectId());
+    adminBuilder.setInstanceId(options.getInstanceId());
+
+    adminBuilder.stubSettings().setEndpoint(options.getAdminHost() + ":" + options.getPort());
+
+    adminBuilder.stubSettings()
+        .setCredentialsProvider(buildCredentialProvider(options.getCredentialOptions()));
+
+    try {
+      ManagedChannel channel = BigtableSession
+          .createChannelPool(options.getAdminHost(), options, options.getChannelCount());
+      adminBuilder.stubSettings().setTransportChannelProvider(
+          FixedTransportChannelProvider.create(GrpcTransportChannel.create(channel)));
+    } catch (GeneralSecurityException ex) {
+      throw new IOException("Could not create channels.", ex);
+    }
+
+    return adminBuilder.build();
+  }
+
+  /**
+   * This method is use to build {@link BatchingSettings}.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param options a {@link BigtableOptions} object.
+   */
+  private static void buildBulkMutationsSettings(Builder builder, BigtableOptions options) {
+    BulkOptions bulkOptions = options.getBulkOptions();
+    BatchingSettings.Builder batchSettingsBuilder = BatchingSettings.newBuilder();
+
+    long autoFlushMs = bulkOptions.getAutoflushMs();
+    long bulkMaxRowKeyCount = bulkOptions.getBulkMaxRowKeyCount();
+    long maxInflightRpcs = bulkOptions.getMaxInflightRpcs();
+    long shortRpcTimeoutMs = options.getCallOptionsConfig().getShortRpcTimeoutMs();
+
+    if (autoFlushMs > 0) {
+      batchSettingsBuilder.setDelayThreshold(ofMillis(autoFlushMs));
+    }
+    FlowControlSettings.Builder flowControlBuilder = FlowControlSettings.newBuilder();
+    if (maxInflightRpcs > 0) {
+      flowControlBuilder
+          .setMaxOutstandingRequestBytes(bulkOptions.getMaxMemory())
+          .setMaxOutstandingElementCount(maxInflightRpcs * bulkMaxRowKeyCount);
+    }
+
+    batchSettingsBuilder
+        .setIsEnabled(bulkOptions.useBulkApi())
+        .setElementCountThreshold(Long.valueOf(bulkOptions.getBulkMaxRowKeyCount()))
+        .setRequestByteThreshold(bulkOptions.getBulkMaxRequestSize())
+        .setFlowControlSettings(flowControlBuilder.build());
+
+    // TODO(rahulkql): implement bulkMutationThrottling & bulkMutationRpcTargetMs, once available
+    builder.bulkMutationsSettings()
+        .setBatchingSettings(batchSettingsBuilder.build())
+        .setSimpleTimeoutNoRetries(
+            ofMillis(shortRpcTimeoutMs));
+  }
+
+  /**
+   * To build BigtableDataSettings#sampleRowKeysSettings with default retries settings.
```

Comment:
To build BigtableDataSettings#sampleRowKeysSettings with default Retry settings. same in line 180, 191, and so forth.